File: 차도희/week2.java

import java.util.LinkedList;
import java.util.Arrays;
public class n42587 {

	public static void main(String[] args) {
		
		class Solution {
		    public int solution(int[] priorities, int location) {
		        int answer = 0;
		        LinkedList<Integer> pri = new LinkedList<Integer>();
		        LinkedList<Integer> index = new LinkedList<Integer>();
                // 주어진 배열과 그 배열의 index를 담는 linkedlist

		        int len = priorities.length;
		        
		        for(int n=0; n<len; n++)
		        {
		            pri.add(priorities[n]);
		            index.add(n);
		        }
		        
		        Arrays.sort(priorities);
                // 우선순위 오름차순 정렬
		        LinkedList<Integer> ans = new LinkedList<Integer>();
		        int i = len-1;
		        
		        while(!pri.isEmpty())
		        {
		            if(pri.get(0) == priorities[i])
		            {
		                ans.add(index.get(0));
                        // 우선순위인 index를 다른 배열에 차례대로 저장해둠
		                i--;
                        // 다음 우선순위로 넘어감
		            }
		            else
		            {
		                int tmp = index.get(0);
		                index.add(tmp);
		                tmp = pri.get(0);
		                pri.add(tmp);
                        // 우선순위가 아니면 맨 뒤로 옮겨줌
		            }
		            index.remove(0);
		            pri.remove(0);
		        }
		        
		        answer = ans.indexOf(location) + 1;
		        return answer;
		    }
		}

        /* test
		Solution sol = new Solution();
		int[] arr = {1,1,9,1,1,1};
		System.out.println(sol.solution(arr, 0));
        */
		
	}
}

File: 이서진/WEEK2.py
def isExistHigherPriority(priorities):
    for p in priorities[1:]:
        if priorities[0] < p:
            return True

    return False


def solution(priorities, location):
    answer = location + 1
    totalNum = len(priorities)

    while len(priorities):  # 다 출력되기 전까지
        if isExistHigherPriority(priorities):  # 검사하는 타겟이 우선순위가 가장 높은 게 아니라면
            priorities.append(priorities[0])  # 맨 뒤로 넣기

            if location == 0:  # 맨 뒤로 보내지는 게 location이라면
                location = len(priorities) - 2  # location 맨 뒤로 변경 (아직 맨 뒤로 넣은 거 삭제 안 된 것 반영)
                answer = totalNum  # 출력 순서 맨 뒤로 변경

            else:  # 맨 앞이 location이 아니라면
                location -= 1  # 위치 앞당겨짐
                answer -= 1  # 출력 순서 앞당겨짐

            del priorities[0]  # 삭제

        else:  # 출력
            if location == 0:  # location의 우선순위가 가장 높다면
                return answer  # 바로 출력이니 정답 리턴

            del priorities[0]  # 그게 아니라면 다시 삭제
            location -= 1  # location 변경
